### PR TITLE
refactor: Collection Create

### PR DESCRIPTION
**Pull request by AnnaEmson**
### What

Refactor CreateCollectionController and CreateCollection into component Create component connected to the store and being only updated by dispatching actions and using hooks. Please be aware the refactoring in this file is not finished yet, as the whole local state managing can be improved further as a lower priority. 

- Write Create.test.js  test following the principles of testing react UI with React testing Library 
- Improve Banner.jsx component test to meet testing principles
- Improve Search.jsx component  tests
- Improve Collections.jsx component  tests
- Connect search to the store 
- Add data-testid to DoubleSelectableBox.jsx  and DoubleSelectableBoxItem.jsx   for better targeting in test
- Add accessible elements to collections search input component
- Remove trailing lines form Input.jsx, DynamicList.jsx and apply some improvements. 
- Move test utilities to utilities folder for better origination 
- Refactor and expand reducer for team fetching for new collection 
- Create new selectors for teams
- Move  fetching teams api to thunks action for separation responsibility and making react component to be only updated  by dispatching actions as per principles. 
- Updated snapshots
- Reduce payload to CollectionEdit.jsx  as only id is needed not the whole team object

### How to review
all test should pass 
technical review of code standards
(please reach out if you have any questions)

### Who can review

front end devs

### CODE REVIEW

File: src/app/components/Input.jsx [hunk 1]
```diff
@@ -136,20 +137,16 @@ export default class Input extends Component {
         return (
             <div className={formClasses}>
                 {!this.props.inline && !this.props.reverseLabelOrder && this.renderLabel()}
-                {this.props.error ? (
+                {this.props.error && (
                     <div id={`input-error-${this.props.id}`} className="error-msg" role="alert">
```
Comment (NoelleDL) [inline src/app/components/Input.jsx:141]: So this statement will be skipped if false?

Comment (AnnaEmson) [reply src/app/components/Input.jsx:141]: yes, exactly, it is conditional rendering in react

**Review by NoelleDL — APPROVED** [review]
LGTM

**Review by dandwal — APPROVED** [review]
Have gone through the code, looks good, have discussed in person som areas and happy that no action is needed on them for this PR, also have been demoed the work and everything looks functional to me so approving.

**Pull request merged**